lib/matplotlib/__init__.py

```diff
@@ -1218,8 +1228,9 @@ def rc_file_defaults():
     with warnings.catch_warnings():
         warnings.simplefilter("ignore", mplDeprecation)
         from .style.core import STYLE_BLACKLIST
-        rcParams.update({k: v for k, v in rcParamsOrig.items()
-                         if k not in STYLE_BLACKLIST})
+        rcParams.update({k: rcParamsOrig[k] for k in rcParamsOrig
+                         if k not in STYLE_BLACKLIST and k != 'backend'})
+        rcParams['backend'] = dict.__getitem__(rcParamsOrig, 'backend')
```

Comment:
I think this is contrary to what the docstring says; if so, the docstring needs an update, and either there or in a comment it would be good to say why this exception is being made.

Reply:
@tacaswell you made this change I think?

Reply:
I think this block can be dropped.